sql/2021/http/http2_http3_usage_by_rank.sql

```diff
@@ -0,0 +1,53 @@
+# standardSQL
+# Percentiles of sites that main page over HTTP/2 or HTTP/3 by rank
+SELECT
+  client,
+  rank_grouping,
+  CASE
+    WHEN rank_grouping = 10000000 THEN 'all'
+    ELSE FORMAT("%'d", rank_grouping)
+  END AS ranking,
+  COUNT(0) AS freq,
+  total,
+  COUNT(0) / total AS pct
+FROM
+  (
+    SELECT
+      client,
+      page,
+      rank
+    FROM
+      `httparchive.almanac.requests`
+    WHERE
+      date = '2021-07-01' AND
+      firstHTML AND
+      LOWER(protocol) IN ('http/2', 'http/3', 'quic', 'h3-29', 'h3-q050')
+  ),
+  UNNEST([1000, 10000, 100000, 1000000, 10000000]) AS rank_grouping
+JOIN
+  (
+    SELECT
+      client,
+      rank_grouping,
+      COUNT(0) AS total
```

Comment:
I think we can expect the total to equal the `rank_grouping` value, or at least very closely. So maybe this isn't necessary?

Reply:
To be honest I like showing total (especially since discovering the UNNEST issue!) - it's a good validation that the total is what it's expected to be!